src/System.Memory/tests/ParsersAndFormatters/SupportedFormats.cs

```diff
@@ -0,0 +1,139 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Collections.Generic;
+
+namespace System.Buffers.Text.Tests
+{
+    // Test metadata that describes a standard format (e.g. 'G', 'D' and 'X') supported by a particular data type.
+    public sealed class SupportedFormat
+    {
+        public SupportedFormat(char symbol, bool supportsPrecision)
+        {
+            Symbol = symbol;
+            SupportsPrecision = supportsPrecision;
+        }
+
+        public char Symbol { get; }
+        public bool SupportsPrecision { get; }
+        public bool IsDefault { get; set; } = false;
+        public bool NoRepresentation { get; set; } = false; // If true, can only be accessed by passing default(StandardFormat). (The weird DateTimeOffset case.)
+        public char FormatSynonymFor { get; set; } = default;
+        public char ParseSynonymFor { get; set; } = default;
+    }
+
+    internal static partial class TestData
```

Comment:
Split into separate file

Reply:
I think it's useful for the SupportedFormats-related metadata to be in the same file as SupportedFormats struct.